packages/calling/calling/src/CallingClient/registration/types.ts

```diff
@@ -1,11 +1,53 @@
-import {WebexRequestPayload} from '../../common/types';
+import {LogContext} from 'Logger/types';
+import {METRIC_EVENT, METRIC_TYPE, REG_ACTION} from '../metrics/types';
+import {IDeviceInfo, MobiusStatus, WebexRequestPayload} from '../../common/types';
+import {CallingClientError} from '../../Errors';
 
 export type Header = {
   [key: string]: string;
 };
 
+export type restoreRegistrationCallBack = (
+  restoreData: IDeviceInfo,
+  loggerContext: LogContext
+) => Promise<boolean>;
+
 export interface IRegistrationClient {
-  createDevice: (url: string) => Promise<WebexRequestPayload>;
-  deleteDevice: (url: string, deviceId: string, deviceUrl: string) => Promise<WebexRequestPayload>;
+  postRegistration: (url: string) => Promise<WebexRequestPayload>;
+  deleteRegistration: (
+    url: string,
+    deviceId: string,
+    deviceUrl: string
+  ) => Promise<WebexRequestPayload>;
   postKeepAlive: (_url: string) => Promise<WebexRequestPayload>;
+
+  setMobiusServers: (primaryMobiusUris: string[], backupMobiusUris: string[]) => void;
+  scheduleFailback429Retry: () => Promise<void>;
+  getRegRetryInterval(attempt: number): number;
+  clearFailbackTimer: () => void;
+  failbackRequired(): boolean;
+  getFailbackInterval(): number;
+  initiateFailback: () => void;
+  startFailbackTimer: (intervalInSeconds: number) => void;
+  startFailoverTimer: () => void;
+  executeFailback: () => Promise<void>;
+  restorePreviousRegistration: (caller: string) => Promise<boolean>;
+  attemptRegistrationWithServers: (caller: string, servers: string[]) => Promise<boolean>;
```

Comment:
suggestion required for better method names.

Reply:
Perhaps triggerRegistration or initRegistration?

Reply:
This function was changed to private. Hence keeping it as attemptRegistrationWithServers. But there is a public method named triggerRegistration as per suggestion. Thanks